release/src/Main.hs

```diff
@@ -47,12 +49,34 @@ main = do
       $logInfo "Reading metadata from pom files"
       artifacts <- liftIO $ mapM (resolvePomData bazelLocations sdkVersion compVersion) artifacts
 
+      let mavenUploadArtifacts = filter (\a -> getMavenUpload $ artMavenUpload a) artifacts
+
+      -- all known targets uploaded to maven
+      let allMavenTargets = Set.fromList $ fmap (T.unpack . getBazelTarget . artTarget) mavenUploadArtifacts
+
+      -- first find out all the missing internal dependencies
+      missingDepsForAllArtifacts <- forM mavenUploadArtifacts $ \a -> do
+          -- run a bazel query to find all internal java and scala library dependencies
+          let bazelQueryCommand = shell $ "bazel query 'kind(\"(scala|java)_library\", deps(" ++ (T.unpack . getBazelTarget . artTarget) a ++ ")) intersect //...'"
+          internalDeps <- liftIO $ lines <$> readCreateProcess bazelQueryCommand ""
+          -- check if a dependency is not already a maven target from artifacts.yaml
+          let missingDeps = filter (`Set.notMember` allMavenTargets) internalDeps
+          return (a, missingDeps)
+
+      let onlyMissing = filter (not . null . snd) missingDepsForAllArtifacts
```

Comment:
Definitely not important here given that this whole script is somewhat adhoc but if you are interested in some Haskell design patterns, I prefer having this kind of filtering change the type, e.g., here you could use something like `mapMaybe (\(a, missing) -> (a,) <$> Data.List.NonEmpty.nonEmpty missing)`. That makes it clearer what guarantees hold afterwards. (Again, I’m totally fine not doing this here).

Reply:
Thanks 👍